pkg/ingester/otlp/convert.go

```diff
@@ -394,11 +394,21 @@ func (p *profileBuilder) convertSampleAttributesToLabelsBack(os *otelProfile.Sam
 }
 
 // convertMappingsBack converts a slice of OpenTelemetry Mapping entries to Google Mapping entries.
-func (p *profileBuilder) convertMappingBack(om *otelProfile.Mapping, dictionary *otelProfile.ProfilesDictionary) (uint64, error) {
+func (p *profileBuilder) convertMappingBack(ol *otelProfile.Location, om *otelProfile.Mapping, dictionary *otelProfile.ProfilesDictionary) (uint64, error) {
 	if i, ok := p.mappingMap[om]; ok {
 		return i, nil
 	}
 
+	hasLines := false
+	hasFunctions := false
+	hasInlineFrames := false
+	for i, line := range ol.Line {
+		hasFunctions = hasFunctions || line.FunctionIndex > 0
+		hasLines = hasLines || line.Line > 0
+		hasInlineFrames = hasInlineFrames || i >= 1
+	}
+	hasFilenames := hasLines
```

Comment:
`hasFilenames`, shouldalso check if there is a non 0 filename attribute on the `function` object:

https://github.com/google/pprof/blob/436353cc1ee64cd00b8975ad18397ca49131f673/proto/profile.proto#L229-L230